Ocean structures

Lazzzycatwastaken opened the pull request:
Added what title says.

![image](https://github.com/user-attachments/assets/e7b88e5f-8d4d-49ee-b2a2-bdffa89a2887)
^ structure in the wild

Lazzzycatwastaken committed:
Sunken Aircraft Carrier structure added.

Lazzzycatwastaken commented:
![image](https://github.com/user-attachments/assets/2dd52b3f-d1d0-4e73-a698-9b4ef76acb60)
oh and structure in the world i built it in.

Lazzzycatwastaken committed:
Oil Rig item pool

Lazzzycatwastaken committed:
Yes i've planted the oil rig malware bob 👿

Lazzzycatwastaken commented:
![image](https://github.com/user-attachments/assets/705cc44b-df4c-4ed2-be01-bded6dee849b)
update

MellowArpeggiation commented:
might need a null spawn weight since nothing else spawns in oceans, potentially making them very common (but I haven't checked myself so iunno)

rivet city gonna go hard

![](https://newfastuff.com/wp-content/uploads/2019/07/7MKuIQu.png)

Lazzzycatwastaken committed:
Added oil rig.
Made all structures rarer (via null spawn weight)

Lazzzycatwastaken committed:
Fixed retardation code
Oilrig building upgraded and made it spawn always on the same Y

Lazzzycatwastaken committed:
forgorred to commit ts

Lazzzycatwastaken committed:
Added beached patrol boat

Lazzzycatwastaken commented:
![image](https://github.com/user-attachments/assets/18854aad-dd37-440d-a7b4-3dbdb11380d0)
boat

HbmMods commented:
Could you send a few more images? A bit hard to tell what's going on from just those few screenshots. A few notes:
* Concrete bricks on a ship feel kinda weird, we might deed new deco blocks for that
* Don't use non-full blocks for the underwater sections, we don't have waterlogging so it looks kinda ugly

Lazzzycatwastaken commented:
1. Well there’s not much I can do about that unless I add the blocks myself, but that should be a new PR. I could take out the carrier for the time being, the other two structures don’t have concrete blocks (apart from the concrete buildings on the oil rig)

2.Again take out the carrier for the time being until more blocks are available.

as for the screenshots im not home so I’m gonna pull out some i sent in the discord before:

![image](https://github.com/user-attachments/assets/2422a3b2-3409-4871-bdfb-4390dc32abfb)
![image](https://github.com/user-attachments/assets/23aace27-ee69-4305-b93c-9ba0f650eb1b)

i dont have more of the beached small boat it has a couple under deck rooms with beds, crates, kitchen etc, and a topside part where the controls are

Lazzzycatwastaken commented:
A thing I also could do with the carrier is replace the concrete with titanium deco blocks since they looked decent enough, idk what about the waterlogging

Lazzzycatwastaken committed:
Fixed boat deleting blocks under it or sum

Lazzzycatwastaken commented:
@HbmMods 
![image](https://github.com/user-attachments/assets/c6d1697e-c62b-4d8d-8d13-fbaffa679b4d)
![image](https://github.com/user-attachments/assets/4aa88489-a31a-4cef-9161-54d3f8b50342)
![image](https://github.com/user-attachments/assets/8454fd2e-1485-4a74-889d-6114bc28c70f)
![image](https://github.com/user-attachments/assets/621ab366-3e2a-4e50-a8f3-962796eb6dc2)
![image](https://github.com/user-attachments/assets/989cc551-b95e-48fe-9950-1aff5843f2bc)
![image](https://github.com/user-attachments/assets/70147817-da17-46f6-93f9-fad63ed95d11)
![image](https://github.com/user-attachments/assets/cd116ac6-d8ce-4a6c-bf94-31b50b49c45b)
![image](https://github.com/user-attachments/assets/676b6514-e158-4587-8ff5-5def8979d058)

Vaern commented:
I'm not too big of a fan of including progression-relevant machines (a derrick is pretty _substantial_ for example) but that oil rig looks great.

The limits of our current deco blocks have been an issue for some time, it's discouraging. Steel deco looks surprisingly good on the rig, though!

Lazzzycatwastaken commented:
wish I could have stuff like "rusted steel deco block" or "broken steel deco block" aswell as some slabs for them, or maybe a different colored ones or some more metal blocks

Lazzzycatwastaken committed:
Merge branch 'HbmMods:master' into AAAAAAAAAAAAAAA

Lazzzycatwastaken committed:
Fixed doors on oil rig,
Fixed waterlogging on aircraft carrier
Balanced loot on aircraft carrier
Removed derrick from oil rig and made it a purely cosmetic build
removed asphalt from carrier because it made you 100x faster in water, and flung you across half the ocean.

Lazzzycatwastaken commented:
Ready for review, builds changed for derrick 
![image](https://github.com/user-attachments/assets/fec31612-fba3-4fd2-ac07-6a2d48e6b505)
Carrier is essentially the same, however no half blocks (apart from barrels, a radio and bobblehead) Also 1.7.10 glass has a weird effect underwater. 
![image](https://github.com/user-attachments/assets/5dabfce8-3458-42c0-b1b7-ab8ff0ccaa78)
![image](https://github.com/user-attachments/assets/b73df09d-d98a-48a4-91cb-d48fdc7a3840)

Lazzzycatwastaken commented:
Oh and the carrier no longer has concrete / bricks on it.

Lazzzycatwastaken commented:
![image](https://github.com/user-attachments/assets/0dea4fa6-7363-40ae-927a-89316ff3f84c)
^ how it looks like (drained)

Lazzzycatwastaken commented:
![image](https://github.com/user-attachments/assets/e7c3987f-70ed-41fc-be6c-6dce0e981cde)
inside
